src/components/SegmentedControl/component.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useCallback } from 'react';
+
+import { Testable, useBuildTestId } from '../../modules/test-ids';
+
+import { Container, Element, Size } from './styled';
+
+export type Props = Omit<React.AllHTMLAttributes<HTMLElement>, 'as' | 'size'> &
+  Testable & {
+    options: string[];
+    size?: Size;
+  };
+
+export const Component = ({
+  children,
+  onMouseEnter,
+  onMouseLeave,
+  onClick,
+  onChange,
+  options,
+  disabled,
+  'data-testid': testId,
+  size = 'huge',
+  ...otherProps
+}: Props) => {
+  const buildTestId = useBuildTestId(testId);
+  const [selected, setSelected] = useState(options[0]);
```

Comment:
I'd rather make this a [controlled component](https://reactjs.org/docs/uncontrolled-components.html) and receive this as a prop. @essj, what do you think?

Reply:
Yeah, that's reasonable in case the parent component needs to know the currently selected option.